app/src/main/java/com/h/pixeldroid/fragments/feeds/FeedFragment.kt

```diff
@@ -72,7 +76,11 @@ open class FeedFragment<T: FeedContent, VH: RecyclerView.ViewHolder?>: Fragment(
 
         swipeRefreshLayout.setOnRefreshListener {
             //by invalidating data, loadInitial will be called again
-            factory.liveData.value!!.invalidate()
+            if (Utils.hasInternet(requireContext())) {
+                factory.liveData.value!!.invalidate()
+            } else {
+                startActivity(Intent(requireContext(), MainActivity::class.java))
```

Comment:
Make sure to clear the previous tasks to disable going back with the back button: `intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK`